### PR TITLE
feat(search): allow max page limit to be configurable

### PR DESCRIPTION
**Pull request by kuangp**
Signed-off-by: Phil Kuang <pkuang@factset.com>

## Hey, I just made a Pull Request!

This change allows integrators to configure the max page limit size instead of being constrained to the current hard coded limit

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/search-backend/config.d.ts [hunk 1]
```diff
@@ -17,6 +17,11 @@
 export interface Config {
   /** Configuration options for the search plugin */
   search?: {
+    /**
+     * Sets the maximum max page limit size. Defaults to 100.
+     */
+    maxPageLimit?: number;
```
Comment (camilaibs) [inline plugins/search-backend/config.d.ts:23]: Does it make sense to configure the default page limit value as well? It is hardcoded to 25 by all search engines, as follows:

https://github.com/backstage/backstage/blob/5cc6d6ce0541fc3a792be2f48caf2f8f43210fc5/plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts#L213

https://github.com/backstage/backstage/blob/5cc6d6ce0541fc3a792be2f48caf2f8f43210fc5/plugins/search-backend-module-pg/src/PgSearchEngine/PgSearchEngine.ts#L140

https://github.com/backstage/backstage/blob/5cc6d6ce0541fc3a792be2f48caf2f8f43210fc5/plugins/search-backend-node/src/engines/LunrSearchEngine.ts#L76

**Review by camilaibs — APPROVED** [review]
Thanks for contributing, looks great to me!
I just asked one question below, if it makes sense, you can also decide to add the other setting to a follow-up pull request 🙂

File: plugins/search-backend/config.d.ts (hunk 1 above)
Comment (kuangp) [reply plugins/search-backend/config.d.ts:23]: ah yeah that's a good idea. I'll make the appropriate updates to expose that option too

Comment (kuangp) [reply plugins/search-backend/config.d.ts:23]: @camilaibs actually I'll do that in a follow up PR if you don't mind. I'm current OOO without my laptop for the week and am hoping to have this in for the next release 🙂

Comment (camilaibs) [reply plugins/search-backend/config.d.ts:23]: Sounds good 🙌🏻

**Review by benjdlambert — APPROVED** [review]
Thanks! :pray:

**Pull request merged**